### DISCUSSION TITLE
Ignore raw .graphql files but still try to import files with extension .graphq.ts

### DISCUSSION THREAD

**Discussion by jgoux** [webpack/webpack#12965] [Q&A] [answered]
Hello all,

I'm using graphql-codegen in my project to generate TypeScript definition types from graphql operations.

I first write my operation as GraphQL SDL in a `MyComponent.graphql` file, and then graphql-codegen generates a `MyComponent.graphql.ts` file containing the types.

What I want now is to import those types without specifying the `.ts` extension : `import { MyAwesomeType } from "./MyComponent.graphql"`.

The issue is that webpack is first attempting to resolve the actual `.graphql` file, so it tells me that I don't have a proper loader to load this file. I'd like webpack to totally ignore all my `.graphql` files and only consider the `.graphql.ts` files so I can import my types as specified above.

It it possible?

Thanks for your help!

**Comment by vankop** [comment]
Hi! Use `resolve.extensions` https://webpack.js.org/configuration/resolve/#resolveextensions

**Reply by jgoux** [reply]
Hello @vankop , this was my first attempt but it doesn't seem to change anything, the `.graphql` file is still attempted before webpack try any `resolve.extensions`.

I think I hit this case from the doc : 
![image](https://user-images.githubusercontent.com/1443499/112186670-aa014d00-8c01-11eb-81bf-fc8f2924f3de.png)

As webpack first check for the file existence, it takes the `.graphql` file in priority over the `.graphql.ts` file. I'd need a config to exclude specific files (all my `.graphql` files) from this check.

Edit : I just discovered the `resolve.restrictions` option and you were the one behind it! https://github.com/webpack/webpack/commit/2f7510abd23f50412eff029fc899a4c61b4cc685

Is it possible to pass a regex saying "anything but files ending with `.graphql`" ? I just tried passing an exhausting list like `config.resolve.restrictions = [/\.(ejs|mjs|js|jsx|ts|tsx|json|cjs)$/];` but then folders aren't resolved (`client/src doesn't match the restriction /\.(ejs|mjs|js|jsx|ts|tsx|json|cjs)$/`)

**Reply by vankop** [reply]
Not sure, `resolve.restrictions = [/^[^\.]+(\.graphql\.ts|\.js|\.ts)?$/]` should work I think..

**Comment by jgoux** [comment, marked as answer]
I've found the right one based on negation! 🎊

`resolve.restrictions = [/^(.(?!.*\.graphql$))+$/]` is matching everything but strings ending with .graphql, problem solved!

Thanks for your help @vankop !
